src/google/protobuf/compiler/java/java_string_field.cc

```diff
@@ -110,7 +112,7 @@ void SetPrimitiveVariables(const FieldDescriptor* descriptor,
     (*variables)["clear_has_field_bit_builder"] = "";
 
     (*variables)["is_field_present_message"] =
-        "!get" + (*variables)["capitalized_name"] + "Bytes().isEmpty()";
+       "!" + (*variables)["isStringEmpty"] + "(" + (*variables)["name"] + "_)";
```

Comment:
Don't we know the type in the generated code such that we could call `(*variables)["name"] + "_.isEmpty()"` here directly?

Reply:
Unfortunately, no.

As I initially described it in https://github.com/protocolbuffers/protobuf/issues/6719, the problem is that (for some reason) fields for **string** type in proto are generated as **java.lang.Object** in Java. So, such fields may contain either **ByteString** or **java.util.String** depending on which representation was requested last time.